openbb_platform/providers/coingecko/openbb_coingecko/utils/helpers.py

```diff
@@ -0,0 +1,146 @@
+"""CoinGecko API Helper Functions."""
+
+import asyncio
+from typing import Any, Dict, List, Optional, Union
+from urllib.parse import urlencode
+
+import requests
+from openbb_core.provider.utils.errors import EmptyDataError
+
+
+class CoinGeckoAPIError(Exception):
+    """CoinGecko API Error."""
```

Comment:
Errors should be a subclass of `OpenBBError` - `from openbb_core.app.model.abstract.error import OpenBBError` - raising these errors would translate into an "Unhandled Exception".

No need to create a custom Error class.